### ISSUE TITLE
Built in fetch method

### ISSUE THREAD

**Issue by jacobwod** [hajkmap/Hajk#204]
# Current status
```
// Reminder on how to check status
$ grep -rwn fetch .
```
## Done: 
:white_check_mark: ./index.js
:white_check_mark:. /models/Click.js
:white_check_mark:. /components/MapSwitcher.js
:white_check_mark:. /plugins/search/SearchModel.js (partly)
:white_check_mark: ./plugins/draw/DrawModel.js

## Left to convert:
:negative_squared_cross_mark: ./plugins/**collector**/CollectorModel.js:61:    fetch(this.url, request)
~~./plugins/layerswitcher/components/MapSwitcher.js:22:    let response = await fetch(~~
:negative_squared_cross_mark: ./plugins/**informative**/InformativeModel.js:64:      response = await fetch(this.url, fetchConfig);
:negative_squared_cross_mark: ./plugins/**search**/SearchModel.js:72:    return fetch(
:negative_squared_cross_mark: ./plugins/search/SearchModel.js:410:    fetch(this.app.config.appConfig.searchProxy + source.url, request).then(
:negative_squared_cross_mark: ./plugins/**edit**/EditModel.js:106:      fetch(src.url, {
:negative_squared_cross_mark: ./plugins/edit/EditModel.js:345:    fetch(url, fetchConfig)
:negative_squared_cross_mark: ./plugins/**simple_layerswitcher**/components/VectorFilter.js:38:    fetch(
:negative_squared_cross_mark: ./plugins/**export**/ExportModel.js:580:    fetch(url, {
:negative_squared_cross_mark: ./plugins/**buffer**/BufferModel.js:150:          fetch(url)
~~./index.js:6:import "abortcontroller-polyfill/dist/polyfill-patch-fetch";~~
~~./index.js:60:fetch("appConfig.json", fetchConfig)~~ // Always local, relative, non-proxy fetch
~~./index.js:81:        fetch("customTheme.json", fetchConfig)~~ // See above 
:negative_squared_cross_mark: ./models/layers/**WMSLayer.js**:139:        fetch(this.proxyUrl + url)
:negative_squared_cross_mark: ./models/layers/**VectorLayer.js**:341:    fetch(url, fetchConfig).then(response => {
:negative_squared_cross_mark: ./models/layers/VectorLayer.js:350:    fetch(url, fetchConfig).then(response => {
:negative_squared_cross_mark: ./components/**Diagram.js**:53:    fetch(url).then(response => {
:negative_squared_cross_mark: ./components/**Table.js**:59:    fetch(url).then(response => {
~~./registerServiceWorker.js:86:  fetch(swUrl, { credentials: "same-origin" })~~

# Original issue
We need to replace all different types of calls (mostly fetch() in `admin` we also find `$.ajax()`) with something local. One way is to expose a local fetch as `this.appModel.fetch()`. This method will only take the URL to call and add "necessary stuff" only once. 

That "necessary stuff" could be:
* An object for the optional `init` parameter of global `fetch()` (see https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch#Parameters)
* Settings for a local proxy 

Both would preferably be read from appConfig.json.

**Comment by jacobwod** [comment]
Created `built-in-fetch` and made a first commit, 67dbc5522b7640b9fb6c6d30356d99875ce2c6a6. 
Still some unresolved design questions. Candidate for 3.x?

**Comment by jacobwod** [comment]
@erikedmark The concept works but we still need to convert plugins, some of which I don't have configured (Diagram/Table/VectorLayer). Feel free to work on `built-in-fetch` branch.

**Comment by jacobwod** [comment]
Stale, closing.

**Comment by jacobwod** [comment]
I'm removing my old branch as it has been superseded by the work done in https://github.com/hajkmap/Hajk/tree/feature/657-fetch-options.
